public/app/features/api-keys/ApiKeysPage.tsx

```diff
@@ -89,15 +96,28 @@ export class ApiKeysPageUnconnected extends PureComponent<Props, State> {
 
   onMigrateApiKeys = async () => {
     try {
-      this.onMigrateAll();
-      let serviceAccountsUrl = '/org/serviceaccounts';
-      locationService.push(serviceAccountsUrl);
-      window.location.reload();
+      const response = await this.props.migrateAll();
+      this.setState((prevState: State) => {
```

Comment:
```suggestion
      this.setState({
        operationSummaryVisible: true,
        operationSummaryData: response,
      });
```

Reply:
talked about it in person. for how to deal with state and we switched to adding it to the storestate